vm_backtrace.c

```diff
@@ -1613,7 +1613,17 @@ rb_profile_frames(int start, int limit, VALUE *buff, int *lines)
                 buff[i] = (VALUE)cfp->iseq;
             }
 
-            if (lines) lines[i] = calc_lineno(cfp->iseq, cfp->pc);
+            if (lines) {
+                // If this frame has a jit_return pointer on it, then the PC
+                // _is not_ reliable.  We can get a crash if we try to calculate
+                // the line number.
+                if (cfp->jit_return) {
+                    lines[i] = 0;
+                }
+                else {
+                    lines[i] = calc_lineno(cfp->iseq, cfp->pc);
```

Comment:
alternatively we could check that `cfp->pc` is within `iseq_encoded ... iseq_encoded+iseq_size` and _sometimes_ get a line number (even though it may be inaccurate or just completely incorrect), but it might be better just to always return 0 indicating that we can't be certain.

Reply:
I was considering this as well.  Maybe we should only return 0 for the topmost frame?  I think all other frames should have a correct PC since it gets flushed before making the call.

(Indeed the crash I saw was in the topmost frame)